patches/net/minecraft/world/item/ProjectileWeaponItem.java.patch

```diff
@@ -1,5 +1,49 @@
 --- a/net/minecraft/world/item/ProjectileWeaponItem.java
 +++ b/net/minecraft/world/item/ProjectileWeaponItem.java
+@@ -24,12 +_,43 @@
+         super(p_43009_);
+     }
+ 
++    /**
++     * @deprecated Use ItemStack sensitive version {@link ProjectileWeaponItem#getSupportedHeldProjectiles(ItemStack)}
++     */
++    @Deprecated
+     public Predicate<ItemStack> getSupportedHeldProjectiles() {
+         return this.getAllSupportedProjectiles();
+     }
+ 
++    /**
++     * If the allowed projectile is weapon stack dependent, return {@code ItemStack.EMPTY}
```

Comment:
Is encouraging people to return empty a good idea here? Wouldn't that break all mods currently using this deprecated method that calls an updated mod's method? Might be better to not say to return empty and let the mod determine if they do want to return empty or some default ammo for legacy callers

Reply:
Sure. Let me change that